File: src/FGVTest.Business/Entities/Pedido.cs
namespace FGVTest.Business.Entities;

public class Pedido
{
    public int CodPedido { get; set; }
    public int CodCliente { get; set; }
    public DateTime DataPedido { get; set; }
    public decimal ValorTotal { get; set; }
    public Cliente Cliente { get; set; } = null!;
    public IEnumerable<ItensPedido> Itens { get; set; } = [];
}

File: src/fgvtest-web/types/pedido.ts
import { Cliente } from './cliente';
import { Produto } from './produto';

export interface ItemPedido {
  produto: Produto;
  quantidade: number;
  precoUnitario: number;
}

export interface Pedido {
  codPedido: number;
  dataPedido: string;
  valorTotal: number;
  cliente: Cliente;
  itens: ItemPedido[];
}

export interface CriarPedidoRequest {
  cnpj: string;
}

export interface AdicionarItemRequest {
  codProduto: number;
  quantidade: number;
}
